modules/httpboot/httpboot_plugin.rb

```diff
@@ -4,11 +4,16 @@ class Plugin < ::Proxy::Plugin
     https_rackup_path File.expand_path("http_config.ru", File.expand_path("../", __FILE__))
 
     plugin :httpboot, ::Proxy::VERSION
-    load_programmable_settings ::Proxy::Httpboot::PluginConfiguration
 
     default_settings :root_dir => '/var/lib/tftpboot'
 
     expose_setting :http_port
     expose_setting :https_port
+
+    after_activation do
+      plugin = ::Proxy::Plugins.instance.find{|p| p[:name] == :httpboot}
+      settings[:http_port] = plugin[:http_enabled] ? Proxy::SETTINGS.http_port : nil
+      settings[:https_port] = plugin[:https_enabled] ? Proxy::SETTINGS.https_port : nil
+    end
```

Comment:
Whether http(s) is enabled or not is not visible in the plugin itself, only in the plugins "wrapper". This must be queried after activation was done, otherwise it's not set.